src/libraries/src.proj

```diff
@@ -9,6 +9,8 @@
              Exclude="@(ProjectExclusions)" />
     <NonNetCoreAppProject Include="@(_allSrc)"
                           Exclude="@(NetCoreAppLibrary->'%(Identity)\src\%(Identity).csproj')" />
+    <NonNetCoreAppProject Include="$(MSBuildThisFileDirectory)*\gen\*.csproj"
```

Comment:
This is different than how #51300 does it. FYI

Reply:
It's the same approach now. Modeled the change there after this one.